decorators/uncontrolled.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+
+export default function uncontrolled (Component, {defaultProp, prop, handler, reset}) {
+  return class Uncontrolled extends Component {
+    componentDidMount () {
+      this.setState({
+        [prop]: this.props[prop] != null
+          ? this.props[prop] || ''
+          : this.props[defaultProp]
+      })
+    }
+
+    handleHandler (handler, e) {
+      if (this.props[prop] != null) {
```

Comment:
Logic can be simpleified. Invert the `if`, put the handler-stuff always in the bottom